### PR TITLE
CalculateAverage_pdrakatos

### PR DESCRIPTION
**Pull request by PanagiotisDrakatos**
#### Check List:
- [x] Tests pass (`./test.sh <username>` shows no differences between expected and actual outputs)
- [x] All formatting changes by the build are committed
- [x] Your launch script is named `calculate_average_<username>.sh` (make sure to match casing of your GH user name) and is executable
- [x] Output matches that of `calculate_average_baseline.sh`
- [x] For new entries, or after substantial changes: When implementing custom hash structures, please point to where you deal with hash collisions (line number)

* Execution time: 00:14.174
* Execution time of reference implementation:00:16.174

### CODE REVIEW

File: calculate_average_pdrakatos.sh [hunk 1]
```diff
@@ -0,0 +1,4 @@
+source "$HOME/.sdkman/bin/sdkman-init.sh"
```
Comment (gunnarmorling) [inline calculate_average_pdrakatos.sh:1]: Please name the file calculate_average_PanagiotisDrakatos.sh, as described in the README.

File: prepare_pdrakatos.sh [hunk 2]
```diff
@@ -0,0 +1,7 @@
+source "$HOME/.sdkman/bin/sdkman-init.sh"
```
Comment (gunnarmorling) [inline prepare_pdrakatos.sh:1]: Same as above.

**Comment by PanagiotisDrakatos** [comment]
@gunnarmorling done please check thanks in advance!!!

File: src/main/java/dev/morling/onebrc/CalculateAverage_PanagiotisDrakatos.java [hunk 3]
```diff
@@ -0,0 +1,231 @@
+package dev.morling.onebrc;
```
Comment (gunnarmorling) [inline src/main/java/dev/morling/onebrc/CalculateAverage_PanagiotisDrakatos.java:1]: Please run `mvn license:format` to add the missing license header.

File: calculate_average_PanagiotisDrakatos.sh [hunk 4]
```diff
@@ -0,0 +1,4 @@
+source "$HOME/.sdkman/bin/sdkman-init.sh"
```
Comment (gunnarmorling) [inline calculate_average_PanagiotisDrakatos.sh:1]: Please make both scripts executable.

**Review by gunnarmorling — CHANGES_REQUESTED** [review]
Please see the comments inline. Also getting this failure now in the native build:

```
Error: Main entry point class 'dev.morling.onebrc.CalculateAverage_pdrakatos' neither found on
classpath: '/home/build/projects/eval/1brc/target/average-1.0.0-SNAPSHOT.jar' nor
modulepath: '/home/build/.sdkman/candidates/java/21.0.1-graal/lib/svm/library-support.jar:/home/build/.sdkman/candidates/java/21.0.1-graal/lib/svm/graal-microservices.jar'.
```

It's a bit of an frustrating experience tbh., it should be verified that everything builds and passes the tests before actually making a submission.

**Comment by PanagiotisDrakatos** [comment]
@gunnarmorling Please check again i added the license and made the scripts executable and ensured that the name of application is valid and updated

**Comment by gunnarmorling** [comment]
Hum, something is off here, there should be no files modified other than those of your entries. Can you please revert all those unrelated changes.

**Comment by PanagiotisDrakatos** [comment]
@gunnarmorling Please check again and sorry for the inconvenience 3 files changed and I merged the latest commits

**Comment by gunnarmorling** [comment]
There's a test failure, please see CI:

```
Validating calculate_average_PanagiotisDrakatos.sh -- src/test/resources/samples/measurements-1.txt

Using java version 21.0.1-graal in this shell.

real	0m0.095s
user	0m0.102s
sys	0m0.024s
0a1
> Kunming;19.8;19.8;19.8

FAILURE: ./test.sh PanagiotisDrakatos failed
Error: Process completed with exit code 1.
```

**Comment by PanagiotisDrakatos** [comment]
@gunnarmorling ok I believe I fixed the problem i checked on my local PC and ran tests with success I had a problem with the prepare sh file. Now it is going to work please check again if possible and thanks in advance :). Wish you an awesome day

**Comment by gunnarmorling** [comment]
There still test failures, see CI.

**Comment by PanagiotisDrakatos** [comment]
@gunnarmorling i pass test_ci.sh with success please check again thanks in advance bro :)

**Comment by gunnarmorling** [comment]
OOMs when running for the 1B file on 32 cores:

```
Exception in thread "main" java.lang.OutOfMemoryError: Cannot reserve 10485782 bytes of direct buffer memory (allocated: 4288681154, limit: 4294967296)
	at java.base/java.nio.Bits.reserveMemory(Bits.java:178)
	at java.base/java.nio.DirectByteBuffer.<init>(DirectByteBuffer.java:127)
	at java.base/java.nio.ByteBuffer.allocateDirect(ByteBuffer.java:360)
	at dev.morling.onebrc.CalculateAverage_PanagiotisDrakatos.concat(CalculateAverage_PanagiotisDrakatos.java:95)
	at dev.morling.onebrc.CalculateAverage_PanagiotisDrakatos$1.next(CalculateAverage_PanagiotisDrakatos.java:72)
	at dev.morling.onebrc.CalculateAverage_PanagiotisDrakatos$1.next(CalculateAverage_PanagiotisDrakatos.java:50)
	at java.base/java.util.Spliterators$IteratorSpliterator.trySplit(Spliterators.java:1920)
	at java.base/java.util.stream.AbstractTask.compute(AbstractTask.java:308)
	at java.base/java.util.concurrent.CountedCompleter.exec(CountedCompleter.java:754)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:387)
	at java.base/java.util.concurrent.ForkJoinTask.invoke(ForkJoinTask.java:667)
	at java.base/java.util.stream.ReduceOps$ReduceOp.evaluateParallel(ReduceOps.java:927)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:233)
	at java.base/java.util.stream.ReferencePipeline.reduce(ReferencePipeline.java:662)
	at dev.morling.onebrc.CalculateAverage_PanagiotisDrakatos.SeekableByteRead(CalculateAverage_PanagiotisDrakatos.java:43)
	at dev.morling.onebrc.CalculateAverage_PanagiotisDrakatos.main(CalculateAverage_PanagiotisDrakatos.java:33)
```

**Comment by PanagiotisDrakatos** [comment]
@gunnarmorling I increased the `-XX:MaxDirectMemorySize` now should be fine please check thanks again bro

**Comment by gunnarmorling** [comment]
Same error as before :(

**Comment by PanagiotisDrakatos** [comment]
@gunnarmorling now i am definitely sure no heap space problem again I changed the formation

**Comment by gunnarmorling** [comment]
Looking good now. 00:24.550.

**Comment by PanagiotisDrakatos** [comment]
@gunnarmorling just one question I am wondering in my local PC, it needs only 5 sec to get executed except the first time which needs 50 sec. Here are my specs:

```
Intel core i9 12900K
Memory DDR5 64GB
Gigabyte Technology Co. Ltd. Z690 AORUS PRO (U3E1)
```

**Pull request merged**